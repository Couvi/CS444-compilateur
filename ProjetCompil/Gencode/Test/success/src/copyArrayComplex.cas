program
	T : array [1 .. 4] of array [1 .. 3] of array [1 .. 2] of integer ;
	i, j : integer;
begin
	for i := 1 to 4 do 
		for j := 1 to 3 do
			T[i][j][1] := i*100+j*10+1;
			T[i][j][2] := i*100+j*10+2;
		end;
    end;
    for i := 1 to 4 do 
		for j := 1 to 3 do
			write("(",T[i][j][1]," ");
			write(T[i][j][2],") ");
		end;
		write("|");
    end;
    new_line;
    write("modif 1");
    new_line;
    for i := 1 to 4 do
    	for j := 1 to 3 do
    		T[i][j][2] := T[i][j][1];
    	end;
    end;
    for i := 1 to 4 do 
		for j := 1 to 3 do
			write("(",T[i][j][1]," ");
			write(T[i][j][2],") ");
		end;
		write("|");
    end;
    new_line;
    write("modif 2");
    new_line;
    T[1][1][1] := 999;
    for i := 1 to 4 do
    	for j := 1 to 3 do
    		T[i][j] := T[i][1];
    	end;
    end;
    for i := 1 to 4 do 
		for j := 1 to 3 do
			write("(",T[i][j][1]," ");
			write(T[i][j][2],") ");
		end;
		write("|");
    end;
    new_line;
    write("modif 3");
    new_line;
    T[1][1][1] := 999;
    for i := 1 to 4 do
    	T[i] := T[1];
    end;
    for i := 1 to 4 do 
		for j := 1 to 3 do
			write("(",T[i][j][1]," ");
			write(T[i][j][2],") ");
		end;
		write("|");
    end;

end.