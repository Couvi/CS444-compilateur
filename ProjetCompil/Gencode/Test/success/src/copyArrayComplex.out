(111 112) (121 122) (131 132) |(211 212) (221 222) (231 232) |(311 312) (321 322) (331 332) |(411 412) (421 422) (431 432) |
modif 1
(111 111) (121 121) (131 131) |(211 211) (221 221) (231 231) |(311 311) (321 321) (331 331) |(411 411) (421 421) (431 431) |
modif 2
(999 111) (999 111) (999 111) |(211 211) (211 211) (211 211) |(311 311) (311 311) (311 311) |(411 411) (411 411) (411 411) |
modif 3
(999 111) (999 111) (999 111) |(999 111) (999 111) (999 111) |(999 111) (999 111) (999 111) |(999 111) (999 111) (999 111) |
